### DISCUSSION TITLE
Need a hand to test Color Mode feature

### DISCUSSION THREAD

**Discussion by sonvnn** [templaza/astroid-framework#316] [General]
Hi @Chacapamac , @Joomlaplates and everyone.

I am working on Color Mode feature. To learn about it please read https://getbootstrap.com/docs/5.3/customize/color-modes/
I almost finished this task and I need help to test it.

### HOW TO TEST:

1. Download developer package https://github.com/templaza/astroid-framework/archive/refs/heads/developer.zip
2. Upload to your Astroid test site via Joomla Installation
3. Go to Plugins > Astroid Plugin > Color Mode > Switch "Yes"
4. Go to Site Template Styles >> Choose one of your Astroid style >> Template Options
5. Go to colors and configure Dark and Light version.
6. Save and go to front-end to see how it look

### EXPECTED RESULTS
You will see Color Mode switcher on the Menu Bar
![Screen Shot 2023-04-13 at 16 51 50](https://user-images.githubusercontent.com/3204103/231723166-83c10bf3-6b66-4e67-9a33-d95fa3ec5920.png)

In the admin you can see 
![Screen Shot 2023-04-13 at 16 54 38](https://user-images.githubusercontent.com/3204103/231723842-67d9efb6-8c57-4e90-9590-30f1b5de30ce.png)
![Screen Shot 2023-04-13 at 16 47 44](https://user-images.githubusercontent.com/3204103/231723643-c36b0f19-b015-4e52-8a03-911be5a670b4.png)

Another example with Color Mode here https://soledaddemo.pencidesign.net/

**Comment by darekmer** [comment]
Hello, will it work with the zero and one template?

**Reply by sonvnn** [reply]
Yes, It works with all Astroid templates include Zero and One.

**Comment by joomlaplates-de** [comment]
Hi Sonny, just tested and it works well in Zero and One template.

**Reply by sonvnn** [reply]
Please help me test with all of your current template. Please report any issue here. I need sure it will work with all environment before release new version :)

Thank you & Best Regards,
Sonny

**Comment by darekmer** [comment]
![1](https://user-images.githubusercontent.com/44267306/231940514-946cdd1a-518d-45af-bc4b-68be2ff7b814.png)

**Reply by sonvnn** [reply]
Remove these lines in your template scss https://github.com/templaza/astroid-framework/blob/one-template/scss/one/components/_pagination.scss#L9:L10

It will work.

Best Regards,
Sonny

**Reply by darekmer** [reply]
I don't want to remove these two lines, it looks ugly. It's better to add a class for 'color-mode'.

**Reply by sonvnn** [reply]
You can add scss to custom.scss
```
@include color-mode(dark) {
  .pagination .page-item {
   &.active .page-link {
      color: $primary;
      background-color: transparent;
      border-color: $primary;
   }
   .page-link {
      color: $dark;
      border: 1px solid $dark;
      &:first-child,
      &:last-child {
         border-radius: 4px;
      }
      &:hover {
         border-color: $primary;
         color: $primary;
         background-color: transparent;
      }
   }
}
}
```

It will work :)

**Reply by darekmer** [reply]
to file: /media/templates/site/child_one/scss/custom.sccs ? If so, it's not working.
![page-link](https://user-images.githubusercontent.com/44267306/232028808-b33c320f-a794-4aff-9222-0a45156e5d65.png)

**Reply by sonvnn** [reply]
I copy from orginal code. You should change $dark value to light color ex: #ffffff;

**Reply by darekmer** [reply]
Sorry, I didn't think. It works ;)

**Reply by sonvnn** [reply]
Let's follow bootstrap documentation. It's just css style for dark mode. Or you can wait until I update one and zero template in the next week :)

**Comment by darekmer** [comment]
After switching to night mode and refreshing the page, it reverts back to the bright mode, but the icon still indicates night mode.

**Reply by sonvnn** [reply]
I fixed this in https://github.com/templaza/astroid-framework/commit/9a748e3e22c0dc8c75c68a4f836ad12cb6605004
Please download developer branch and reinstall it again. Please clear cache to recompile scss.

Thanks & Best Regards,
Sonny

**Reply by Chacapamac** [reply]
> After switching to night mode and refreshing the page, it reverts back to the bright mode, but the icon still indicates night mode.

**No revert to bright mode on my test website** —> http://joom4-astro.graf-test.ca/
 —> **But** when refreshing, while in **dark mode**, I can see the **default bright mode** for a fraction of second before the dark mode appear.

**Reply by sonvnn** [reply]
@Chacapamac Please update your index.php of your template follow this one https://github.com/templaza/astroid-framework/blob/one-template/index.php

It will help template recognize color mode in the first load.

Best Regards,
Sonny

**Reply by Chacapamac** [reply]
@sonvnn  — Yep, with the new **index.php**, no more flashing of the Bright mode when refreshing in Dark Mode

**Comment by Chacapamac** [comment]
Testing right now!, Coming back soon!

**Reply by Chacapamac** [reply]
**@sonvnn the feature seem to wok great, no console error.**

The only thing I can see is when switching the color toggler to “**Dark**” mode the container _(astroid-header-section)_ slightly change by loosing some pixels vertically _(shrink by few pixels)_ 

![gif-animated](https://user-images.githubusercontent.com/1907276/232058549-01bde613-5924-4e8e-a42e-5db3225800b5.gif)

•The only structural changes between the two modes in the CSS of **“input class="form-check-input switcher"** are:
**LIGHT**
![Screen Shot 2023-04-14 at 9 36 58 AM](https://user-images.githubusercontent.com/1907276/232058829-5a979f37-f2d6-4e89-9fc7-1e48b4f85cc7.jpg)

**DARK**
![Screen Shot 2023-04-14 at 9 37 36 AM](https://user-images.githubusercontent.com/1907276/232058973-697f0e08-6d25-4b9e-baf7-a01bfb1a18f3.jpg)

**and I loose the logo?**
![loose-logo](https://user-images.githubusercontent.com/1907276/232059958-d712f865-529f-4192-bef1-33569fbc8d26.gif)

**You can test here** —> http://joom4-astro.graf-test.ca/

**Reply by sonvnn** [reply]
Please clear astroid cache and try it again! 

Let's tell me how you feel about it then ;)

**Reply by Chacapamac** [reply]
@sonvnn 

**For the logo disappearing in Dark mode I find the problem** ———————————————————————
The **Dark Mode** add the ability to change the logo to adapt to the new colors
— In the **Astroid Template Manager** —> **Header settings**

**_Suggestion to help users in this case_**
I think You should add a note in the  **Astroid Template Manager** —> **Colors** —> **Header** & **Sticky Header** like...

You can change the main logo image to adapt to Your **Dark** & **Light mode** (default) background colors choice in the main **Header**(link to) menu

_— You should have a link to the main “Header” menu here_

Another note under ALL **Dark Mode** logo choice in the Main **Header** menu, maybe you can have a note like.

You can change the logo section background color for *Light mode** (default) or **Dark mode** in the main menu **Colors** —> **Header**(link to) & **Sticky Header**(link to)




**The slight move in the structure persist????** ———————————————————————
- Tested in Firefox and Chrome

**• Displacement of few pixel on structure** (loosing 1 to 2 pixel in height in Dark Mode)

**Comment by darekmer** [comment]
![boot-4](https://user-images.githubusercontent.com/44267306/232422608-68293b58-c40d-41e2-9bab-917769d0dfca.png)

**Comment by Chacapamac** [comment]
@sonvnn, Just put “**Bootstrap**”

**Do we have access to the Full Color Palette of Bootstrap 5 ?** —> https://getbootstrap.com/docs/5.0/customize/color/

**Comment by darekmer** [comment]
Can anything be done about it?
Light iFrame background transparent
![light](https://user-images.githubusercontent.com/44267306/232704606-fcec87df-ec03-4b46-a0d5-215bd3417b0c.png)
Dark iFrame background white, no transparent ;(
![dark](https://user-images.githubusercontent.com/44267306/232704740-c7c10ea8-5b7d-4604-a86b-652d1f256fed.png)

**Reply by sonvnn** [reply]
May you send me your website url? I can inspect your css code and give your some solution.

Best Regards,
Sonny

**Reply by darekmer** [reply]
only test for localhost

**Reply by darekmer** [reply]
maybe the iframe code will help:
iframe 1:
```
<p><a class="weatherwidget-io" href="https://forecast7.com/pl/50d1120d07/prusy/" data-label_1="Prusy" data-label_2="Pogoda" data-days="3">Prusy Pogoda</a></p>
<p>
<script>
!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src='https://weatherwidget.io/js/widget.min.js';fjs.parentNode.insertBefore(js,fjs);}}(document,'script','weatherwidget-io-js');
</script>
</p>
```
iframe 2:
```
<p></p>
<!-- Airly Widget Begin -->
<p>
<script src="https://airly.org/map/airly.js" type="text/javascript"></script>
</p>
<p><iframe style="width: 100%; height: 140px; border: none;" id="airly_2092721151" src="https://airly.org/widget/v2/?&amp;height=140&amp;displayMeasurements=false&amp;displayCAQI=true&amp;autoHeight=false&amp;autoWidth=true&amp;language=pl&amp;indexType=AIRLY_CAQI&amp;unitSpeed=metric&amp;unitTemperature=celsius&amp;latitude=50.113379&amp;longitude=20.085078&amp;locationId=11337"></iframe></p>
<!-- Airly Widget End -->
```

**Reply by darekmer** [reply]
Firefox generates a white background for dark mode on iframes. ;(
Interestingly, Firefox correctly generates a transparent background on iframes with a light/dark toggle in index.html using Bootstrap 5.
However, in Chrome and Edge, the same issue occurs after refreshing the page.

**Comment by Chacapamac** [comment]
@darekmer — I just ry on my test website —> http://joom4-astro.graf-test.ca/
You right 
![Screen Shot 2023-04-23 at 12 57 30 PM](https://user-images.githubusercontent.com/1907276/233855026-02ea687e-be8a-4cd1-a206-2272b039829c.jpg)

> Firefox generates a white background for dark mode on iframes.

I try to add `allowtransparency="true"` but that don’t work either

**Reply by Chacapamac** [reply]
Try again and the white background is added only in “Dark Mode”
I put another color background for the “Light Mode” and the Iframe background is correctly transparent.

**Reply by darekmer** [reply]
The background in dark mode should be transparent, while in light mode it is transparent as expected. However, in dark mode, the background appears as white. If you locate your iframe element using F12 and remove the 'color-scheme: dark;' property in [data-bs-theme='dark'] in the CSS, the white background disappears.

**Reply by Chacapamac** [reply]
This should not happen and should be fix.

**Comment by darekmer** [comment]
Can be added to custom.css and then the background is transparent in dark mode:
[data-bs-theme="dark"] {
	color-scheme: inherit;
}

**Comment by darekmer** [comment]
In the Categories module, the active link appears with a dark color in dark mode.
![link color](https://user-images.githubusercontent.com/44267306/235129397-37747926-989b-4832-84b4-f945a4a3a0a5.png)

**Comment by sonvnn** [comment]
Hi All,

My latest project with Color Mode feature https://express.jollyany.co/
Please help me check it.

Thanks & Best Regards,
Sonny

**Comment by darekmer** [comment]
Nice project. Found, maybe it's meant to be?
![pagination](https://github.com/templaza/astroid-framework/assets/44267306/133eabbc-5570-4b39-a180-c3e7b4ffef4f)
![hover](https://github.com/templaza/astroid-framework/assets/44267306/927be738-2081-4a6e-96f4-d5072f0a7773)

**Reply by sonvnn** [reply]
Thank you! It's Bootstrap missing style. I fixed it in https://github.com/templaza/astroid-framework/commit/d3aec95fa7b3176fc64ac88589c17aaf09ce61c5

Best Regards,
Sonny
